libs/estdlib/src/gen_server.erl

```diff
@@ -326,8 +335,20 @@ call(Name, Request, TimeoutMs) when is_atom(Name) ->
             call(Pid, Request, TimeoutMs)
     end;
 call(Pid, Request, TimeoutMs) when is_pid(Pid) ->
-    Ref = erlang:make_ref(),
-    call_internal(Pid, {'$call', {self(), Ref}, Request}, TimeoutMs).
+    MonitorRef = monitor(process, Pid),
+    Pid ! {'$call', {self(), MonitorRef}, Request},
+    receive
+        {'DOWN', MonitorRef, process, Pid, normal} ->
+            {error, noproc};
+        {'DOWN', MonitorRef, process, Pid, Reason} ->
+            erlang:exit(Reason);
+        {MonitorRef, Reply} ->
+            demonitor(MonitorRef, [flush]),
+            Reply
+    after TimeoutMs ->
+        demonitor(MonitorRef, [flush]),
+        erlang:exit(timeout)
```

Comment:
Could we update the test (line 203) to assert semantics match OTP?

Reply:
See instead line 132 of the test (which addresses the estdlib/OTP discrepancy)